sdk/core/azure-core/src/http/retry_policy.cpp

```diff
@@ -94,16 +94,17 @@ Context::Key const RetryKey;
 int32_t RetryPolicy::GetRetryCount(Context const& context)
 {
   int32_t number = -1;
-  if (!context.TryGetValue<int32_t>(RetryKey, number))
+  int32_t* ptr = &number;
+  if (!context.TryGetValue<int32_t*>(RetryKey, ptr))
   {
     // Context with no data abut sending request with retry policy = -1
     // First try = 0
     // Second try = 1
     // third try = 2
     // ...
-    number = -1;
+    *ptr = -1;
```

Comment:
With the new guarantee that we won't touch the value, we don't need any more assignments to the pointer. it is already associated with an int that equals to -1.
BTW, I'd change the value we store by `RetryKey` to "`int32 const*`" - we don't expect calling code to modify it.

Reply:
> BTW, I'd change the value we store by `RetryKey` to "`int32 const*`" - we don't expect calling code to modify it.

Unrelated change. I will leave it to @vhvb1989 to do so separately. If you want to do it @antkmsft, go for it :)